internal/cmdcommon/prompt/oneofstringlist_test.go

```diff
@@ -0,0 +1,72 @@
+package prompt_test
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/kyma-project/cli.v3/internal/cmdcommon/prompt"
+	"github.com/stretchr/testify/require"
+)
+
+func TestOneOfStringListPrompt_Table(t *testing.T) {
+	tests := []struct {
+		name       string
+		input      string
+		values     []string
+		parseFunc  func(string) (string, error)
+		want       string
+		wantErr    string
+		wantOutput string
+	}{
+		{
+			name:       "Valid input",
+			input:      "banana\n",
+			values:     []string{"apple", "banana", "orange"},
+			want:       "banana",
+			wantErr:    "",
+			wantOutput: "Select a fruit:\n - apple\n - banana\n - orange\n\nType the version number: ",
+		},
+		{
+			name:       "Invalid input",
+			input:      "Faraon Ramzes XIII\n",
+			values:     []string{"apple", "banana", "orange"},
+			want:       "",
+			wantErr:    "provided value is not present on the list: Faraon",
+			wantOutput: "Select a fruit:\n - apple\n - banana\n - orange\n\nType the version number: ",
+		},
+		{
+			name:       "Empty input",
+			input:      "\n",
+			values:     []string{"apple", "banana", "orange"},
+			want:       "",
+			wantErr:    "no value was selected",
+			wantOutput: "Select a fruit:\n - apple\n - banana\n - orange\n\nType the version number: ",
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			input := bytes.NewBufferString(tc.input)
```

Comment:
We can test the case when the Read method returns an error. For example, to cover the case when the user runs the `add` command on the CI/CD system, where STDIN is not configured. To do so, I will suggest using the `ErrReader` function from [this package](https://go.dev/src/testing/iotest/reader.go).